src/app/clusters/descriptor/descriptor.cpp

```diff
@@ -99,17 +99,18 @@ CHIP_ERROR DescriptorAttrAccess::ReadDeviceAttribute(EndpointId endpoint, Attrib
 
 CHIP_ERROR DescriptorAttrAccess::ReadClientServerAttribute(EndpointId endpoint, AttributeValueEncoder & aEncoder, bool server)
 {
-    CHIP_ERROR err = aEncoder.EncodeList([&endpoint, server](const TagBoundEncoder & encoder) -> CHIP_ERROR {
-        uint16_t clusterCount = emberAfClusterCount(endpoint, server);
+    CHIP_ERROR err =
+        aEncoder.EncodeList([&endpoint, server](const AttributeValueEncoder::ListEncodeHelper & encoder) -> CHIP_ERROR {
```

Comment:
I wonder whether this might have ended up more readable and maintanable if we had done:
```suggestion
        aEncoder.EncodeList([&endpoint, server](auto encoder) -> CHIP_ERROR {
```
from the start in all of these consumers, so the actual type is not relevant as long as it has the APIs we expect.... thoughts?

Reply:
Yes, checked the comment for EncodeList, and confirmed the expected ListEncodeHelper only relies on the Encode() function.